### PR TITLE
[PBNTR-180] Advanced Table Kit: Rails

### PR DESCRIPTION
**Pull request by nidaqg**
**What does this PR do?**

Creates v0.1 of the rails version of the Advanced Table kit:

- ✅ Use Table kit under the hood, render table with the table_data prop (same as tableData on react side)
- ✅ Kit to iterate over nested data to any depth and render subrows with appropriate indentation/styling
- ✅ column_definitions to take same syntax as columnDefinitions on react side. 
- ✅ Create header and body subcomponents
- ✅ Apply appropriate classnames to get the same look as react table
- ✅ Render first column based on data and column_definitions, in similar way as react side
- ✅ Render row level toggle buttons (no interactivity yet) only render if row has children)
- ✅ enable_toggle_expansion prop 
- ✅ Write methods for row depth to give us same data that Tanstack gives us for row.depth 
- ✅ All subrows rendered on page as first step. Expansion/collapse will come with implementation of turbo
- ✅ Table to take children subcomponents but also work without it

[Runway Story](https://nitro.powerhrg.com/runway/backlog_items/PBNTR-180)

**Screenshots:** 

Screenshot 2024-03-19 at 8 42 23 AM

### CODE REVIEW

File: playbook-website/app/helpers/advanced_table_example_mock_data.rb [hunk 1]
```diff
@@ -0,0 +1,280 @@
+# frozen_string_literal: true
+
+TABLE_DATA = [
```
Comment (nidaqg) [inline playbook-website/app/helpers/advanced_table_example_mock_data.rb:3]: I didn't want to put this giant Array inside the doc example, so was looking for the best place for it. It's just for our doc example and does not need to ship with the kit. Is this place ok for this file? @jasperfurniss

File: playbook/app/pb_kits/playbook/pb_advanced_table/docs/_advanced_table_default.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<% require './app/helpers/advanced_table_example_mock_data.rb' %>
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_advanced_table/docs/_advanced_table_default.html.erb:1]: I'd love to expose this like we did for react

File: playbook/app/pb_kits/playbook/pb_advanced_table/_advanced_table.scss [hunk 3]
```diff
@@ -82,7 +82,8 @@
 
   // Vertical separator
   .table-header-cells:first-child,
-  td:first-child {
+  td:first-child,
+  .pb_table_td:first-child {
     box-shadow: 1px 0px 0px 0px #e4e8f0 !important;
```
[the remark this answers is not in the document]
Comment (nidaqg) [reply playbook/app/pb_kits/playbook/pb_advanced_table/_advanced_table.scss:87]: I can make this a variable, probably just got missed when we first put the kit in Playbook

File: playbook/app/pb_kits/playbook/pb_advanced_table/docs/_advanced_table_default.html.erb (hunk 2 above)
Comment (nidaqg) [reply playbook/app/pb_kits/playbook/pb_advanced_table/docs/_advanced_table_default.html.erb:1]: Sure, we can chat about it!

File: playbook-website/app/helpers/advanced_table_example_mock_data.rb (hunk 1 above)
Comment (nidaqg) [reply playbook-website/app/helpers/advanced_table_example_mock_data.rb:3]: updating for transparency: discussed with Jason, moving or changing that data file will be covered on a follow up PR, not a blocker on this one.

**Review by thestephenmarshall — APPROVED** [review]

**Pull request merged**